### ISSUE TITLE
Serializers without pk field do not trigger subscription callbacks

### ISSUE THREAD

**Issue by gegenschall** [theY4Kman/dcrf-client#2]
I'm trying to subscribe to a specific model instance using something similar to this:
```javascript
const taskSubscription = this.$dcrf.subscribe('url', this.id, () => {
    console.log('called!')
})
```

I can see that a subscribe request is sent
```json
{
  "stream":"url",
  "payload":{
    "action":"subscribe_instance",
    "request_id":"25f8361e-c627-4b1e-8142-e0214b8b3999",
    "pk":"some-pk-value"
  }
}
```

and the server sends a reply as follows:
```json
{
  "stream":"url",
  "payload":{
    "errors":[

    ],
    "data":null,
    "action":"subscribe_instance",
    "response_status":201,
    "request_id":"25f8361e-c627-4b1e-8142-e0214b8b3999"
  }
}
```

Updates are sent down by the server similar to this message:
```json
{
  "stream":"task-status",
  "payload":{
    "errors":[
    ],
    "data":{
--- snip ---
    },
    "action":"update",
    "response_status":200,
    "request_id":"25f8361e-c627-4b1e-8142-e0214b8b3999"
  }
}
```

To me it seems the messages are correctly formatted, yet the callback is never triggered. It seems that this only happens for `update` requests as `delete` requests correctly trigger the callback. 

edit: There seem to be several issues at play here. 
1. `request_id` seem to be changing between the `payload` and `selector` in the `FifoDispatcher` so they are never `isMatch`-ed to each other. Generating `request_id`s before calling `subscribe` does solve this issue.
2. A general design issue: DCRFClient uses `pk` as the lookup value when REST framework will usually use `id`. Adding `pk` *and* `id` to `Meta.fields` solves this in a hacky way.

**Comment by theY4Kman** [comment]
Oof, yeah, the hardcoded `pk` is awful. That selector building really ought to be exposed and customizable. I'm looking at the code now, and there are so many direct references to `DCRFClient.xyz`, so it doesn't even seem like it can be overridden in a subclass, either :facepalm:

Full disclosure: I'm not sure if I'd be able to fix things this week, but _maybe_ this weekend. Though, if you can find your way around the codebase, I can be quick with PR review and releases. I figured I'd just expose a `buildSubscriptionListener` option in the config, much like `preprocessPayload` — though, in the long-term, I think it makes sense to have a separate interface for building selectors & formatting messages, so everything about the protocol could be customized.

**Comment by theY4Kman** [comment]
Hmm, so an update on this (sorry for the super extended delay). It appears that after subscribing to a, object, updates to the object are passed through the serializer before being sent over websocket — so, if `id` is used instead of `pk`, this is reflected in the update event. However, if the object is deleted, it cannot be run through the serializer (due to its fatal lack of existence :skull:), and when sent over websocket, `pk` is always used.

Subscriptions in the client are currently implemented with a single selector. My plan is to split subscriptions into two selectors: one for updates, and one for deletes. The unfortunate side-effect of this would be that handlers for deletion would have to reference `pk` instead of `id`.

I've been debating whether to retain this weird idiosyncrasy of the server-side for the sake of purity, or whether to always ensure the handler payload contains the correct `id` field. I'm leaning toward correcting the idiosyncrasy by default, and letting the user disable this behaviour, if so desired.

---

While on this discovery and fixing path, I've decided to make a few other changes:

 - `buildMultiplexedMessage`, `buildRequestResponseSelector`, `buildSubscribeSelector`, and `buildSubscribePayload` may now be overridden in the client options. This'll allow a high degree of control, for situations with extreeeeeme customization

 - Though I said otherwise in https://github.com/theY4Kman/dcrf-client/issues/6#issuecomment-737528387, I actually _did_ end up adding a `pkField` option, because I really didn't like the amount of copy-pasting needed just to change `pk` in `buildSubscribeSelector`. Since discovering that delete event idiosyncrasy, though, I find it's imperative in order to support seamless update/delete handlers for subscriptions.

**Comment by theY4Kman** [comment]
@befocken @gegenschall I've opened #7 to address this issue. It needs some documentation before merging, but the change should afford the ability to pass `client = dcrf.connect(..., { pkField: 'id' })` and all will be well. I added integration tests with a serializer defining `fields = '__all__'` to ensure it works with that, as well.

Also, I updated the integration tests to use the latest DCRF, which supports Channels 3.
